components/Seo.php

```diff
@@ -100,22 +106,32 @@ private function getDataFromStaticPage()
             // If RainLab.Translate plugin is installed we need to ensure, that
             // lang prefix is not in our way, so we are removing it from url.
             if (PluginManager::instance()->hasPlugin('RainLab.Translate')) {
-                if (Str::contains($url, '/')) {
-                    $url = explode('/', $url)[1];
-                }
+                // Remove language prefix from URL to get clean URL
+                // Simply exploding the URL after first slash does not work for 
+                // multilevel websites
+                $url = substr($url, 3); 
+
+                // If we're in root then use root as URL
+                if (!strlen($url))
+                    $url = '/';
             }
 
             $router = new Router(Theme::getActiveTheme());
             $page = $router->findByUrl($url);
-            $viewBag = $page->getViewBag();
-
-            $this->title = $viewBag->meta_title ?? $viewBag->title;
-            $this->description = $viewBag->meta_description;
-            $this->keywords = $viewBag->seo_keywords;
-            $this->canonicalUrl = $viewBag->seo_canonical_url ?? Request::url();
-            $this->redirectUrl = $viewBag->seo_redirect_url;
-            $this->robotsIndex = $viewBag->seo_robots_index;
-            $this->robotsFollow = $viewBag->seo_robots_follow;
+            // Proper way of obtaining localized/translated viewBag data from 
+            // Static page. According to documentation the getViewBag() method 
+            // is for internal use only.
+            $viewBag = $page->viewBag;
+
+            // Updated to reflect change from $page->getViewBag() to $page->viewBag 
+            // which is an array now
+            $this->title = $viewBag['meta_title'] ?? $viewBag['title'];
+            $this->description = $viewBag['meta_description'] ?? NULL;
+            $this->keywords = $viewBag['seo_keywords'] ?? NULL;
+            $this->canonicalUrl = $viewBag['seo_canonical_url'] ?? Request::url();
+            $this->redirectUrl = $viewBag['seo_redirect_url'] ?? NULL;
+            $this->robotsIndex = $viewBag['seo_robots_index'] ?? NULL;
+            $this->robotsFollow = $viewBag['seo_robots_follow'] ?? NULL;
```

Comment:
Please change `NULL` to `null` as [PSR-2](https://www.php-fig.org/psr/psr-2/) requires keywords to be in lowercase